src/main/java/hudson/plugins/analysis/core/HealthAwareRecorder.java

```diff
@@ -342,15 +372,14 @@ protected Object readResolve() {
     }
 
     @Override
-    public final boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher,
-            final BuildListener listener) throws InterruptedException, IOException {
+    public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener)
```

Comment:
Strict binary compatibility is broken here... again, not sure if this is the correct thing to do or not